cmd/celvet/main.go

```diff
@@ -81,23 +91,55 @@ func main() {
 	}
 
 	limitErrors := celvet.CheckMaxLimits(structural)
+	costErrors, compileErrors, otherErrors := celvet.CheckExprCost(structural)
+	if useJSON {
+		emitJSON(limitErrors, costErrors, compileErrors, otherErrors)
+	} else {
+		emitText(limitErrors, costErrors, compileErrors, otherErrors, *humanReadable)
+	}
+
+	if len(limitErrors)+len(costErrors)+len(compileErrors) > 0 {
+		os.Exit(1)
+	}
+}
+
+func emitJSON(limitErrors []*celvet.LimitError, costErrors []*celvet.CostError, compileErrors []*celvet.CompilationError, otherErrors []error) {
+	type JSONOutput struct {
+		LimitErrors   []*celvet.LimitError       `json:"limitErrors,omitempty"`
+		CostErrors    []*celvet.CostError        `json:"costErrors,omitempty"`
+		CompileErrors []*celvet.CompilationError `json:"compileErrors,omitempty"`
+		OtherErrors   []string                   `json:"otherErrors,omitempty"`
+	}
+
+	encoder := json.NewEncoder(os.Stdout)
+	encoder.SetEscapeHTML(false)
+	strOtherErrors := make([]string, 0)
```

Comment:
Since the length of `strOtherErrors` will be the same as `otherErrors`, it is more idiomatic to use `make([]string, len(otherErrors)` and then  fill the array directly (no append).

Reply:
That makes a lot more sense, thanks - fixed